apps/cms.c

```diff
@@ -885,10 +892,32 @@ int cms_main(int argc, char **argv)
             goto end;
     }
 
-    in = bio_open_default(infile, 'r',
-                          binary_files ? FORMAT_BINARY : informat);
-    if (in == NULL)
-        goto end;
+    if (digesthex != NULL) {
+        if (infile != NULL) {
+            BIO_printf(bio_err, "Cannot use -digest without -in\n");
```

Comment:
The error should be the other way around shouldn't it? `Cannot use -digest with -in`

Reply:
That's true. I reverted this change.